temporal-sdk/src/main/java/io/temporal/workflow/DynamicUpdateHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2023 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.workflow;
+
+import io.temporal.common.converter.EncodedValues;
+
+/**
+ * Use DynamicUpdateHandler to process any update dynamically. This is useful for a library level
+ * code and implementation of DSLs.
+ *
+ * <p>Use {@link Workflow#registerListener(Object)} to register an implementation of the
+ * DynamicUpdateListener. Only one such listener can be registered per workflow execution.
+ *
+ * <p>When registered any signals which don't have a specific handler will be delivered to it.
+ *
+ * @see DynamicQueryHandler
+ * @see DynamicSignalHandler
+ * @see DynamicWorkflow
+ */
+public interface DynamicUpdateHandler {
+  void validate(String updateName, EncodedValues args);
+
+  EncodedValues execute(String updateName, EncodedValues args);
```

Comment:
This should be called `handle` to be consistent with other "handlers"

Really though I think `DynamicSignalHandler` and `DynamicQueryHandler` got it a bit wrong. They should have qualified their method names to `handleDynamicSignal` and `handleDynamicQuery` respectively. Then this should have methods named `validateDynamicUpdate` and `handleDynamicUpdate`. So that way I could implement all three interfaces in the same class without ambiguity.

Reply:
> You still can through overloading no?

I don't believe so. For example, given the `DynamicQueryHandler.handle` and `DynamicSignalHandler.handle` methods define the same name and same arg types but different return types, I cannot think of a way for a single Java class to implement both (my Java may be rusty though).